api/client/modelupgrader/upgrader.go

```diff
@@ -80,7 +80,7 @@ func (c *Client) UpgradeModel(
 }
 
 // UploadTools uploads tools at the specified location to the API server over HTTPS.
-func (c *Client) UploadTools(r io.ReadSeeker, vers version.Binary) (tools.List, error) {
+func (c *Client) UploadTools(_ context.Context, r io.ReadSeeker, vers version.Binary) (tools.List, error) {
```

Comment:
This is fixed in the next PR - we need to remove various api struct context attributes :-(